### DISCUSSION TITLE
Can you please make a DB Condition Handeling Project?

### DISCUSSION THREAD

**Discussion by daninus14** [ultralisp/ultralisp#244] [Ideas]
Using `mito` and `dbi` in both production and development causes issues because of threads calling the same connection and a lack of restart of the connection if it's lost among others.

See here for an example: https://stackoverflow.com/questions/77087733/common-lisp-mito-db-error-connection-to-database-server-lost

From [here](https://github.com/fukamachi/mito/issues/40) it looks like you've addressed this issues in [db.lisp](https://github.com/ultralisp/ultralisp/blob/master/src/db.lisp#L67-L88).

Can you please make that into it's own project available in quicklisp or ultralisp for everyone? 

Can you please write docs? In particular:

- How would you use it? Always wrap a db call `mito:some-action` with a `with-connection` macro?
- When to use `with-connection` vs `with-transaction` 
- When to use `with-lock` which seems to only be used [here](https://github.com/ultralisp/ultralisp/blob/0a9cf955bd44fc3bcf27c0a31d50fc81ed0078c0/src/pipeline/checking.lisp#L28) even though it's also imported in [base.lisp](https://github.com/ultralisp/ultralisp/blob/0a9cf955bd44fc3bcf27c0a31d50fc81ed0078c0/src/downloader/base.lisp#L6)

I think everyone will benefit greatly from this!

Thanks!

**Comment by svetlyak40wt** [comment]
I've already extracted these utilities into https://github.com/40ants/40ants-pg
However didn't found a time to document it yet.

**Reply by svetlyak40wt** [reply]
Created an issue: https://github.com/40ants/40ants-pg/issues/1

But not sure when I can add docs – now all my free time spending on a new project and a new UI kit for [Reblocks](https://github.com/40ants/reblocks).

**Reply by daninus14** [reply]
Thanks, that's awesome, I'll look into the project you already started 😄 all the best with Reblocks!
